internal/pkg/manifest/svc.go

```diff
@@ -39,7 +40,85 @@ type Service struct {
 
 // ServiceImage represents the service's container image.
 type ServiceImage struct {
-	Build *string `yaml:"build"` // Path to the Dockerfile.
+	Build BuildArgsOrString `yaml:"build"` // Path to the Dockerfile.
+}
+
```

Comment:
dumb question: Instead of multiple methods is there a reason why we can't do: `func (s *ServiceImage) Build() BuildArgsOrString`

Reply:
I left it this way to prevent mixing docker abstractions across too many packages. Returning the `manifest.BuildArgsOrString` struct here means that the `cli` package has to convert between `manifest.BuildArgsOrString` and `docker.BuildArguments` in `svc deploy` explicitly. I thought it was slightly better (less confusing to read) to just have `cli` assemble the `docker.BuildArguments` struct from these basic types like string and map.

Reply:
Gotcha, I think the two are pretty similar. Right now the `cli` package still does the transformation work but it's with the `cli.dockerParams` struct which introduces a new struct to the pkg scope.

I think it's fine to do transformations within procedures (i.e the dependency package is not surfaced in the public api of the `cli` package). 

Actually, instead of returning `BuildArgsOrString` what do you think of returning `BuildArgs`? and the function
`function(s *ServiceImage) Build() BuildArgs` can do the work that's inside the current `Dockerfile` method to assign the appropriate value?  
This way the `cli` pkg wouldn't have to do that logic

Reply:
This is now implemented in a `(s *ServiceImage) BuildConfig(wsRoot) *docker.BuildArguments` method which extracts the work from `svc deploy` to pkg `manifest`. 

I've also added specific errors that get thrown when unmarshaling invalid build configurations (no dockerfile specified, no supported fields specified)

Reply:
Looks good! I am not sure about returning a `docker.BuildArguments` instead of a `manifest.DockerBuildArgs` from the `manifest` pkg, but I'm okay with keeping it as is.